akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.http.caching.javadsl
+
+import akka.actor.ActorSystem
+import akka.annotation.DoNotInherit
+import akka.http.caching.scaladsl.{ CachingSettingsImpl, LfuCacheSettingsImpl }
+
+import scala.concurrent.duration.Duration
+
+/**
+ * Public API but not intended for subclassing
+ */
+@DoNotInherit
+abstract class CachingSettings private[http] () { self: CachingSettingsImpl ⇒
+  def lfuCacheSettings: LfuCacheSettings
```

Comment:
This seems a bit awkward since we only have `LfuCacheSettings` for now... it does seem the most future-proof thing to do though.

Reply:
Yeah, it's for future proofing... call me paranoid ;)

Reply:
For extra paranoid-ness we should also add `require(...)` calls here or in `LfuCacheSettingsImpl` to ensure positive capacity values.